packages/db/src/collection.ts

```diff
@@ -536,8 +537,21 @@ export class CollectionImpl<
       return // Already started or in progress
     }
 
+    // Enter loading state immediately so callers see progress
     this.setStatus(`loading`)
 
+    // If a previous cleanup is still in flight, wait for it to finish before actually starting
+    if (this.cleanupInFlight) {
+      this.cleanupInFlight.then(() => {
```

Comment:
This method is called `startSync` but is no longer really synchronous. If there is a cleanup in flight then this function will return before we actually started. As a caller of `startSync` i would expect the collection to be started which is no longer the case. Should we make this function async `private async start(): Promise<void>` ?